File: frontend/lib/models/daily_card_entry.dart
class DailyCardEntry {
  final int id;
  final int showroomId;
  final String? showroomName;
  final int userId;
  final String? userName;
  final int cardAccountId;
  final String? bankName;
  final String? lastFour;
  final String entryDate;
  final double amount;
  final String? notes;
  final bool isLocked;
  final String? createdAt;
  final String? updatedAt;

  DailyCardEntry({
    required this.id,
    required this.showroomId,
    this.showroomName,
    required this.userId,
    this.userName,
    required this.cardAccountId,
    this.bankName,
    this.lastFour,
    required this.entryDate,
    required this.amount,
    this.notes,
    required this.isLocked,
    this.createdAt,
    this.updatedAt,
  });

  factory DailyCardEntry.fromJson(Map<String, dynamic> json) {
    return DailyCardEntry(
      id: json['id'] as int,
      showroomId: json['showroom_id'] as int,
      showroomName: json['showroom']?['name'] as String?,
      userId: json['user_id'] as int,
      userName: json['user']?['name'] as String?,
      cardAccountId: json['card_account_id'] as int,
      bankName: json['card_account']?['bank_name'] as String?,
      lastFour: json['card_account']?['last_four'] as String?,
      entryDate: json['entry_date'] as String,
      amount: double.tryParse(json['amount'].toString()) ?? 0.0,
      notes: json['notes'] as String?,
      isLocked: json['is_locked'] == true || json['is_locked'] == 1,
      createdAt: json['created_at'] as String?,
      updatedAt: json['updated_at'] as String?,
    );
  }
}

File: ios/Runner/Models/DailyCardEntry.swift
import Foundation

struct DailyCardEntry: Decodable, Identifiable {
    let id: Int
    let showroomId: Int
    let showroomName: String?
    let userId: Int
    let userName: String?
    let cardAccountId: Int
    let bankName: String?
    let lastFour: String?
    let entryDate: String
    let amount: Double
    let notes: String?
    let isLocked: Bool
    let createdAt: String?
    let updatedAt: String?

    var displayCard: String {
        guard let b = bankName, let l = lastFour else { return "Card" }
        return "\(b) •••• \(l)"
    }

    private enum CodingKeys: String, CodingKey {
        case id
        case showroomId    = "showroom_id"
        case showroom
        case userId        = "user_id"
        case user
        case cardAccountId = "card_account_id"
        case cardAccount   = "card_account"
        case entryDate     = "entry_date"
        case amount, notes
        case isLocked      = "is_locked"
        case createdAt     = "created_at"
        case updatedAt     = "updated_at"
    }

    private struct NameNested: Decodable { let name: String }
    private struct CardNested: Decodable {
        let bankName: String?
        let lastFour: String?
        enum CodingKeys: String, CodingKey {
            case bankName = "bank_name"
            case lastFour = "last_four"
        }
    }

    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        id            = try c.decode(Int.self, forKey: .id)
        showroomId    = try c.decode(Int.self, forKey: .showroomId)
        userId        = try c.decode(Int.self, forKey: .userId)
        cardAccountId = try c.decode(Int.self, forKey: .cardAccountId)
        entryDate     = try c.decode(String.self, forKey: .entryDate)
        notes         = try? c.decode(String.self, forKey: .notes)
        createdAt     = try? c.decode(String.self, forKey: .createdAt)
        updatedAt     = try? c.decode(String.self, forKey: .updatedAt)

        showroomName = (try? c.decode(NameNested.self, forKey: .showroom))?.name
        userName     = (try? c.decode(NameNested.self, forKey: .user))?.name

        let card = try? c.decode(CardNested.self, forKey: .cardAccount)
        bankName = card?.bankName
        lastFour = card?.lastFour

        let amtRaw = try c.decode(AnyCodable.self, forKey: .amount)
        amount = Double("\(amtRaw.value)") ?? 0

        if let b = try? c.decode(Bool.self, forKey: .isLocked) {
            isLocked = b
        } else {
            isLocked = (try? c.decode(Int.self, forKey: .isLocked)) == 1
        }
    }
}
